### PR TITLE
Update request/proof response

### PR DESCRIPTION
**Pull request by taobun**
Add request/proof response struct

### CODE REVIEW

File: chain/cmd/bandsv/main.go [hunk 1]
```diff
@@ -15,7 +17,53 @@ type OracleRequest struct {
...
+type GetRequestResp struct {
+	Request zoracle.RequestWithReport `json:"request"`
+	Proof   Proof                     `json:"proof"`
+}
+
+type IAVLPath struct {
+	Height        uint64 `json:"height"`
+	Size          uint64 `json:"size"`
+	Version       uint64 `json:"version"`
+	IsDataOnRight bool   `json:"isRight"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: `json:"isDataOnRight"`

File: chain/cmd/bandsv/main.go [hunk 2]
```diff
@@ -15,7 +17,53 @@ type OracleRequest struct {
...
+	SiblingHash   []byte `json:"siblingHash"`
+}
+
+type TMSignature struct {
+	R                []byte `json:"r"`
+	S                []byte `json:"s"`
+	V                uint   `json:"v"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: `uint8`

File: chain/cmd/bandsv/main.go [hunk 3]
```diff
@@ -15,7 +17,53 @@ type OracleRequest struct {
...
+	SignedDataSuffix []byte `json:"suffix"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: `signedDataSuffix`

File: chain/cmd/bandsv/main.go (hunk 2 above)
Comment (prin-r) [inline chain/cmd/bandsv/main.go]: I don't like "uint".  Please specify "uint" what. (uint8, uint16, ...)

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/bandsv/main.go [hunk 4]
```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
...
+	VersionAndChainIdHash       []byte `json:"versionAndChainIdHash"`
+	TimeHash                    []byte `json:"timeHash"`
+	TxCountAndLastBlockInfoHash []byte `json:"txCountAndLastBlockInfoHash"`
+	ConsensusDataHash           []byte `json:"consensusDataHash"`
+	LastResultsHash             []byte `json:"lastResultsHash"`
+	EvidenceAndProposerHash     []byte `json:"evidenceAndProposerHash"`
+}
+
+type BlockHashProof struct {
+	ZoracleRootHash        []byte                 `json:"zoracleRootHash"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: `OracleIAVLStateHash`

File: chain/cmd/bandsv/main.go [hunk 5]
```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
...
+	OtherStoreHash         []byte                 `json:"otherStoreHash"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: `OtherStoresMerkleHash`

File: chain/cmd/bandsv/main.go [hunk 6]
```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
...
+	R                []byte `json:"r"`
+	S                []byte `json:"s"`
+	V                uint8  `json:"v"`
+	SignedDataSuffix []byte `json:"signedDataSuffix"`
+}
+
+type Proof struct {
+	BlockHeight      uint64           `json:"blockHeight"`
+	ZoracleTreeProof ZoracleTreeProof `json:"zoracleTreeProof"`
+	BlockHashProof   BlockHashProof   `json:"blockHashProof"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: Is it ok to have field name the same as struct name?

File: chain/cmd/bandsv/main.go [hunk 7]
```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
...
+
+type BlockHashProof struct {
+	ZoracleRootHash        []byte                 `json:"zoracleRootHash"`
+	OtherStoreHash         []byte                 `json:"otherStoreHash"`
+	BlockHeaderMerkleParts BlockHeaderMerkleParts `json:"blockHeaderMerkleParts"`
+	SignedDataPrefix       []byte                 `json:"signedDataPrefix"`
+	Signatures             []TMSignature          `json:"signatures"`
+}
+
+type ZoracleTreeProof struct {
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: Let's name this `OracleDataProof`

File: chain/cmd/bandsv/main.go [hunk 8]
```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
...
+type BlockHeaderMerkleParts struct {
+	VersionAndChainIdHash       []byte `json:"versionAndChainIdHash"`
+	TimeHash                    []byte `json:"timeHash"`
+	TxCountAndLastBlockInfoHash []byte `json:"txCountAndLastBlockInfoHash"`
+	ConsensusDataHash           []byte `json:"consensusDataHash"`
+	LastResultsHash             []byte `json:"lastResultsHash"`
+	EvidenceAndProposerHash     []byte `json:"evidenceAndProposerHash"`
+}
+
+type BlockHashProof struct {
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: Let's name this `BlockRelayProof`

File: chain/cmd/bandsv/main.go [hunk 9]
```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
...
 	CodeHash  []byte `json:"codeHash"`
+}
+
+type GetRequestResp struct {
+	Request zoracle.RequestWithReport `json:"request"`
+	Proof   Proof                     `json:"proof"`
+}
+
+type IAVLPath struct {
+	SubtreeHeight  uint64 `json:"subtreeHeight"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: `uint8`

File: chain/cmd/bandsv/main.go [hunk 10]
```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
 type OracleRequestResp struct {
 	RequestId uint64 `json:"id"`
 	CodeHash  []byte `json:"codeHash"`
+}
+
+type GetRequestResp struct {
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: `OracleInfoResp`

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/bandsv/main.go [hunk 11]
```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
...
+	ZoracleRootHash        []byte                 `json:"zoracleRootHash"`
+	OtherStoreHash         []byte                 `json:"otherStoreHash"`
+	BlockHeaderMerkleParts BlockHeaderMerkleParts `json:"blockHeaderMerkleParts"`
+	SignedDataPrefix       []byte                 `json:"signedDataPrefix"`
+	Signatures             []TMSignature          `json:"signatures"`
+}
+
+type ZoracleTreeProof struct {
+	LeafNode LeafNode   `json:"valueNode"`
+	Paths    []IAVLPath `json:"paths"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: Name this `MerklePaths`

File: chain/cmd/bandsv/main.go [hunk 12]
```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
...
+type BlockHashProof struct {
+	ZoracleRootHash        []byte                 `json:"zoracleRootHash"`
+	OtherStoreHash         []byte                 `json:"otherStoreHash"`
+	BlockHeaderMerkleParts BlockHeaderMerkleParts `json:"blockHeaderMerkleParts"`
+	SignedDataPrefix       []byte                 `json:"signedDataPrefix"`
+	Signatures             []TMSignature          `json:"signatures"`
+}
+
+type ZoracleTreeProof struct {
+	LeafNode LeafNode   `json:"valueNode"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: IMO We don't need `LeafNode` struct. Just pack all fields here.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/bandsv/main.go (hunk 6 above)
Comment (taobun) [reply chain/cmd/bandsv/main.go]: ok

File: chain/cmd/bandsv/main.go [hunk 13]
```diff
@@ -15,7 +17,57 @@ type OracleRequest struct {
...
+	SignedDataPrefix       []byte                 `json:"signedDataPrefix"`
+	Signatures             []TMSignature          `json:"signatures"`
+}
+
+type OracleDataProof struct {
+	Version     uint64     `json:"version"`
+	RequestId   uint64     `json:"requestId"`
+	CodeHash    []byte     `json:"codeHash"`
+	Params      []byte     `json:"params"`
+	Value       []byte     `json:"value"`
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: Let's call this `Data`. It can theoretically contains multiple values.

File: chain/cmd/bandsv/main.go [hunk 14]
```diff
@@ -15,7 +17,57 @@ type OracleRequest struct {
...
 	CodeHash  []byte `json:"codeHash"`
-	Params    []byte `json:"params"`
+}
+
+type OracleInfoResp struct {
+	Request zoracle.RequestWithReport `json:"request"`
+	Proof   Proof                     `json:"proof"`
+}
+
+type IAVLPath struct {
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: `IAVLMerklePath`, according to evm/bridge code.

**Review by sorawit — CHANGES_REQUESTED** [review]
Overall looking good. Just a few more comments.

**Review by sorawit — APPROVED** [review]

**Pull request merged**